test/integration/apiserver/coordinated_leader_election_test.go

```diff
@@ -39,27 +39,94 @@ import (
 	apiservertesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
 	"k8s.io/kubernetes/pkg/controlplane/apiserver"
 	"k8s.io/kubernetes/test/integration/framework"
+	"k8s.io/utils/ptr"
 )
 
 func TestSingleLeaseCandidate(t *testing.T) {
+	tests := []struct {
+		name                   string
+		preferredStrategy      v1.CoordinatedLeaseStrategy
+		expectedHolderIdentity *string
+	}{
+		{
+			name:                   "default strategy, has lease identity",
+			preferredStrategy:      v1.OldestEmulationVersion,
+			expectedHolderIdentity: ptr.To("foo1"),
+		},
+	}
 	featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.CoordinatedLeaderElection, true)
-
 	server, err := apiservertesting.StartTestServer(t, apiservertesting.NewDefaultTestServerOptions(), nil, framework.SharedEtcd())
 	if err != nil {
 		t.Fatal(err)
 	}
 	defer server.TearDownFn()
 	config := server.ClientConfig
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			cletest := setupCLE(config, ctx, cancel, t)
+			defer cletest.cleanup()
+			go cletest.createAndRunFakeController("foo1", "default", "foo", "1.20.0", "1.20.0", tc.preferredStrategy)
+			cletest.pollForLease(ctx, "foo", "default", tc.expectedHolderIdentity)
+		})
+	}
+}
 
+func TestSingleLeaseCandidateUsingThirdPartyStrategy(t *testing.T) {
+	tests := []struct {
+		name                   string
+		preferredStrategy      v1.CoordinatedLeaseStrategy
+		expectedHolderIdentity *string
+	}{
+		{
+			name:              "third party strategy, no holder identity",
+			preferredStrategy: v1.CoordinatedLeaseStrategy("foo.com/bar"),
+			// Because a third-party CoordinatedLeaseStrategy is in use,
+			// the coordinated leader election controller doesn't manage
+			// the lease's leader election and does not set the holder identity,
+			// and leave it to some other controller. But fall
```

Comment:
"fall through" is used in our controller but we don't have to state that implementation detail. Just say the lease will be created with the strategy set but holderIdentity set to nil.